client/editor/tool_tile.cpp

```diff
@@ -86,17 +137,224 @@ void editor_tool_tile::close_tool()
+      struct extra_type *res = ptile->resource;
+      ui.value_resource->setText(extra_name_translation(res));
+    } else {
+      ui.value_resource->setText(_("None"));
+    }
+
+    // tile road (highest level)
+    ui.value_road->setText(get_tile_road_name(ptile));
+    // tile infrastructure
+    ui.value_infra->setText(get_tile_infra_name(ptile));
+    // tile base
+    ui.value_base->setText(get_tile_base_name(ptile));
+    // tile hut
+    ui.value_hut->setText(get_tile_hut_name(ptile));
+    // tile nuisance (pollution, fallout)
+    ui.value_nuisance->setText(get_tile_nuisance_name(ptile));
   }
-  clear_hover_state();
+}
+
+/**
+ * \brief Return a string of the final road on a tile
+ */
+const QString editor_tool_tile::get_tile_road_name(struct tile *ptile)
+{
+  QString sroad = nullptr;
+  extra_type_by_cause_iterate(EC_ROAD, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sroad = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sroad != nullptr) {
+    return sroad;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a strong of the added infrastructure on a tile
+ */
+const QString editor_tool_tile::get_tile_infra_name(struct tile *ptile)
+{
+  QString sinfra = nullptr;
+  extra_type_by_cause_iterate(EC_IRRIGATION, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sinfra = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  extra_type_by_cause_iterate(EC_MINE, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      sinfra + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (sinfra != nullptr) {
+    return sinfra;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a strong of the any nuisances on a tile
+ */
+const QString editor_tool_tile::get_tile_nuisance_name(struct tile *ptile)
+{
+  QString snuisance = nullptr;
+  extra_type_by_cause_iterate(EC_POLLUTION, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      snuisance = extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  extra_type_by_cause_iterate(EC_FALLOUT, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      snuisance + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (snuisance != nullptr) {
+    return snuisance;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a string of any huts on a tile
+ */
+const QString editor_tool_tile::get_tile_hut_name(struct tile *ptile)
+{
+  QString shut = nullptr;
+  extra_type_by_cause_iterate(EC_HUT, pextra)
+  {
+    if (tile_has_extra(ptile, pextra)) {
+      shut + extra_name_translation(pextra);
+    }
+  }
+  extra_type_by_cause_iterate_end;
+
+  if (shut != nullptr) {
+    return shut;
+  } else {
+    return "None";
+  }
+}
+
+/**
+ * \brief Return a string of any bases on a tile
+ */
+const QString editor_tool_tile::get_tile_base_name(struct tile *ptile)
```

Comment:
This doesn't seem to work on a fogged tile with a fortress.

Reply:
I opened an issue for this